src/studio/src/designer/backend.Tests/Controllers/TextsControllerTests.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Configuration;
+using Altinn.Studio.Designer.Controllers;
+using Altinn.Studio.Designer.Services.Interfaces;
+using Designer.Tests.Mocks;
+using Designer.Tests.Utils;
+
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Designer.Tests.Controllers
+{
+    public class TextsControllerTests : IClassFixture<WebApplicationFactory<TextsController>>
+    {
+        private readonly WebApplicationFactory<TextsController> _factory;
+        private readonly string _versionPrefix = "designer/api/v2";
+
+        public TextsControllerTests(WebApplicationFactory<TextsController> factory)
+        {
+            _factory = factory;
+            TestSetupUtils.SetupDirtyHackIfLinux();
+        }
+
+        [Fact]
```

Comment:
A bit too much line breaks perhaps? If you group the lines according to the test arrangement pattern Arrange, Act, Assert it's easier to read and to see which part of the method that just setup (arrange), which part that actually tests the code (Act) and which part the verifies the result (Assert).

```csharp
        [Fact]
        public async Task GetText_ReturnsNorwegianBokmalText()
        {
            // Arrange
            HttpClient client = GetTestClient();
            string dataPathWithData = $"{_versionPrefix}/ttd/new-resource-format/texts/nb";
            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, dataPathWithData);
            await AuthenticationUtil.AddAuthenticateAndAuthAndXsrFCookieToRequest(client, httpRequestMessage);

            // Act
            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
            response.EnsureSuccessStatusCode();
            string responseBody = await response.Content.ReadAsStringAsync();
            JsonDocument responseDocument = JsonDocument.Parse(responseBody);
            Dictionary<string, string> responseDictionary = JsonSerializer.Deserialize<Dictionary<string, string>>(responseDocument.RootElement.ToString());

            // Assert
            Dictionary<string, string> expectedDictionary = new Dictionary<string, string>
            {
                { "nb_key1", "nb_value1" },
                { "nb_key2", "nb_value2" }
            };
            Assert.Equal(expectedDictionary, responseDictionary);
        }
```